examples/chat2/chat.go

```diff
@@ -77,16 +82,19 @@ func NewChat(ctx context.Context, node *node.WakuNode, connNotifier <-chan node.
 			chat.C = theFilter.C
 		}
 	} else {
-		sub, err := node.Relay().Subscribe(ctx)
-		if err != nil {
-			chat.ui.ErrorMessage(err)
-		} else {
-			chat.C = make(chan *protocol.Envelope)
-			go func() {
-				for e := range sub.Ch {
-					chat.C <- e
-				}
-			}()
+
+		for _, topic := range topics {
+			sub, err := node.Relay().SubscribeToTopic(ctx, topic)
+			if err != nil {
+				chat.ui.ErrorMessage(err)
+			} else {
+				chat.C = make(chan *protocol.Envelope)
+				go func() {
+					for e := range sub.Ch {
+						chat.C <- e
+					}
+				}()
+			}
```

Comment:
Shouldn't the wait-group be based on number of times we loop?
It is hardcoded to 7 in line 101.

Reply:
That 7 refers to the goroutines created from line 102 to 113:
```
go chat.parseInput()
go chat.receiveMessages()
go chat.welcomeMessage()
go chat.connectionWatcher(&connectionWg, connNotifier)
go chat.staticNodes(&connectionWg)
go chat.discoverNodes(&connectionWg)
go chat.retrieveHistory(&connectionWg)
```

Reply:
Ah..got it.